packages/integration-tests/src/entities/Author.ts

```diff
@@ -155,39 +142,34 @@ export class Author extends AuthorCodegen {
   });
 
   /** Example of an async property that returns an entity. */
-  readonly latestComment2: AsyncProperty<Author, Comment | undefined> = hasReactiveAsyncProperty(
-    { publisher: "comments", comments: {} },
-    (author) => author.publisher.get?.comments.get[0].entity ?? author.comments.get[0].entity,
-  );
+  readonly latestComment2: AsyncProperty<Author, Comment | undefined> = hasReactiveAsyncProperty({
+    publisher: "comments",
+    comments: {},
+  }, (author) => author.publisher.get?.comments.get[0].entity ?? author.comments.get[0].entity);
 
   /** Example of an async property that has a conflicting/overlapping reactive hint with ^. */
-  readonly allPublisherAuthorNames: AsyncProperty<Author, string | undefined> = hasReactiveAsyncProperty(
-    { publisher: { authors: "firstName" } },
-    (author) => author.publisher.get?.authors.get.flatMap((a) => a.firstName).join(),
-  );
+  readonly allPublisherAuthorNames: AsyncProperty<Author, string | undefined> = hasReactiveAsyncProperty({
+    publisher: { authors: "firstName" },
+  }, (author) => author.publisher.get?.authors.get.flatMap((a) => a.firstName).join());
```

Comment:
Hrm, this reformatting might be problematic, b/c I think it's undoing whatever prettier/dprint would do...

Hm, I wonder if this is just a legitimate difference between prettier & dprint--like dprint is "prettier-ish", but not 100% the same, and we generally keep them separated out, like `*Codegen.ts` is done by dprint (fast b/c we regularly format all files) & `*.ts` is done by prettier (not fast, but joist-codegen only touches them once on file creation, so it doesn't matter).

But, I bet if we more regularly "ran dprint on prettier-controlled files", or "ru nprettier on dprint-controlled files", we'd get these sorts of things...

Makes me wonder if each integration needs to know "is my source file usually formatted by dprint or by prettier" and then just use the appropriate formatter...

Kinda annoying, but given you'd added caching for the doc-gen piece, we should be good to use prettier I think...everyone's very first `yarn joist-codegen` run will re-prettier the entity files to get their `.cache` populated, but that shouldn't be too bad...

Wdyt of like conditional dprint-vs-prettier formatting of the output?

Reply:
Mentioned it above, I'll try out recast because I'm thinking if we can get perfect comment insertion without touching the rest of the line, we should be able to avoid all this.

Reply:
Cool! Fwiw a little curious to see how recast handles it, b/c even if it parses/outputs the existing code with 100% accuracy (which will be great), we'll still be inserting net-new/changed code, which will probably? have to go through a formatter? Maybe?

Or maybe if recast keeps everything basically as-is, and only our ~few lines of new comments are what "need formatting", then it won't matter as much whether we use prettier or dprint? Not sure...